### PR TITLE
Add precision (by default) to Height/Weight Pickers

### PR DESCRIPTION
**Pull request by eschramm**
Closes https://github.com/CareEvolution/RKStudio-Participant/issues/1099. This is a simple approach to add precision mostly as discussed in https://github.com/CareEvolution/Consumers/issues/11101. 

ResearchKit currently does not have a high-precision concept for Height and for Weight and USC, it currently uses ounces.

While still allowing support for "low precision" (current behavior), the accompanying changes in [CEVResearchKit](https://github.com/CareEvolution/CEVResearchKit/pull/270) will now default to this new "high precision" mode.

|Mode|BEFORE|AFTER|
|---|---|---|
|**Height - USC**|Height - USC - Low|Height - USC - High|
|**Height - Metric**|Height - Metric - Low|Height - Metric - High|
|**Weight - USC**|Weight - USC - Low|Weight - USC - High|
|**Weight - Metric**|Weight-Metric-Low|Weight - Metric - High|

### Testing
- Check to see that for a **Question Step** or **Form Step** that uses an **Answer Format** of **Height**  shows ability to 
select fraction of inches as shown above
- Check to see that for a **Question Step** or **Form Step** that uses an **Answer Format** of **Weight**  shows ability to select fraction of pounds as shown above

### CODE REVIEW

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h [hunk 1]
```diff
@@ -1459,20 +1459,57 @@ ORK1_CLASS_AVAILABLE
...
+ @param measurementSystem       The measurement system to use. See `ORK1MeasurementSystem` for the
+                                    accepted values.
+ @param numericPrecision        The numeric precision used by the picker. If you pass
+                                    `ORK1NumericPrecisionDefault`, the picker will use 1 cm
+                                    increments for the metric measurement system and 1 ft and 1 inch
+                                    increments for the USC measurement system, which mimics the
+                                    default iOS behavior. If you pass `ORK1NumericPrecisionLow`, the
+                                    picker will use 1 cm increments for the metric measurement
+                                    system and whole pound increments for the USC measurement
+                                    system. If you pass `ORK1NumericPrecisionHigher`, the picker
```
Comment (mmertsock) [inline ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h]: Typo (here and various other places): "ORK1NumericPrecisionHigher" should be "ORK1NumericPrecisionHigh"

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h [hunk 2]
```diff
@@ -1459,20 +1459,57 @@ ORK1_CLASS_AVAILABLE
...
+ 
+ @param measurementSystem       The measurement system to use. See `ORK1MeasurementSystem` for the
+                                    accepted values.
+ @param numericPrecision        The numeric precision used by the picker. If you pass
+                                    `ORK1NumericPrecisionDefault`, the picker will use 1 cm
+                                    increments for the metric measurement system and 1 ft and 1 inch
+                                    increments for the USC measurement system, which mimics the
+                                    default iOS behavior. If you pass `ORK1NumericPrecisionLow`, the
+                                    picker will use 1 cm increments for the metric measurement
+                                    system and whole pound increments for the USC measurement
```
Comment (mmertsock) [inline ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h]: "whole pound increments"?

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h [hunk 3]
```diff
@@ -1459,20 +1459,57 @@ ORK1_CLASS_AVAILABLE
...
+                                    system. If you pass `ORK1NumericPrecisionHigher`, the picker
+                                    use 0.1 cm increments for the metric measurement system,
+                                    and 0.1 inch increments for the USC measurement system.
+ 
+ @return An initialized weight answer format.
```
Comment (mmertsock) [inline ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h]: Typo: "weight" instead of "height"

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m [hunk 4]
```diff
@@ -2910,17 +2918,9 @@ - (NSString *)stringForAnswer:(id)answer {
...
-                NSString *poundsString = [formatter stringFromNumber:@(pounds)];
-                answerString = [NSString stringWithFormat:@"%@ %@", poundsString, ORK1LocalizedString(@"MEASURING_UNIT_LB", nil)];
-            } else {
-                double pounds, ounces;
-                ORK1KilogramsToPoundsAndOunces(((NSNumber *)answer).doubleValue, &pounds, &ounces);
-                NSString *poundsString = [formatter stringFromNumber:@(pounds)];
-                NSString *ouncesString = [formatter stringFromNumber:@(ounces)];
-                answerString = [NSString stringWithFormat:@"%@ %@, %@ %@", poundsString, ORK1LocalizedString(@"MEASURING_UNIT_LB", nil), ouncesString, ORK1LocalizedString(@"MEASURING_UNIT_OZ", nil)];
-            }
+            double pounds = ORK1KilogramsToPounds(((NSNumber *)answer).doubleValue);
```
Comment (mmertsock) [inline ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m:2921]: The `pounds` value here will [always be rounded to a whole integer](https://github.com/CareEvolution/ResearchKit/blob/8cfb94763af40d5fa208f3598104a84c7ee89117/ORK1Kit/ORK1Kit/Common/ORK1Helpers_Internal.h#L328) — not what we want, right?

File: ORK1Kit/ORK1Kit/Common/ORK1WeightPicker.m [hunk 5]
```diff
@@ -131,16 +131,16 @@ - (void)setAnswer:(id)answer {
             }
             [_pickerView selectRow:poundsIndex inComponent:0 animated:NO];
         } else {
-            double pounds, ounces;
-            ORK1KilogramsToPoundsAndOunces(((NSNumber *)answer).doubleValue, &pounds, &ounces);
+            double pounds = ORK1KilogramsToPounds(((NSNumber *)answer).doubleValue);
```
Comment (mmertsock) [inline ORK1Kit/ORK1Kit/Common/ORK1WeightPicker.m:134]: Same comment here about ORK1KilogramsToPounds?

**Review by mmertsock — COMMENTED** [review]
Overall design and implementation looks good, just a few questions/typos

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m (hunk 4 above)
Comment (eschramm) [reply ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m:2921]: Good catch, now that double from `ORK[1]KilogramsToPounds` is unrounded

**Review by mmertsock — APPROVED** [review]
LGTM 🚢

**Pull request merged**